lib/libc/armstdc/include/string.h

```diff
@@ -0,0 +1,19 @@
+#ifndef ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_STRING_H_
+#define ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_STRING_H_
+
+#include <stddef.h>
+#include <zephyr/toolchain.h>
+
+#include_next <string.h>
```

Comment:
include_next depends on careful header path construction; this seems fragile to me.

Reply:
Not sure how else to do this w/o duplicating the contents of `<string.h>` from the toolchain.  Which probably has its own set of issues.  I think this is the best choice at this point.

Reply:
agreed. Best of a bad situation pasting POSIX compatibility on top of a non-POSIX C library.